### PR TITLE
Add a note about tc.unquote.decl/def verbosity to user manual

### LINKED ISSUE #6963: Keep user-facing debug printing for reflection
Currently debug printing is disabled unless you compile with `-f debug`. However, there are a few cases where we want to expose debug printing to the user without recompiling Agda:

- We export a `debugPrint` reflection primitive
- Printing the code generated by `unquoteDecl` is sometimes very important for debugging reflection code.

We should change these to use the always-on debug printing.

### PR DESCRIPTION
**Pull request by UlfNorell**
Closes #6963

### CODE REVIEW

**Review by andreasabel — APPROVED** [review]
Thanks!

Rendering here (note at the bottom of this section): https://agda--6981.org.readthedocs.build/en/6981/language/reflection.html#id3

File: doc/user-manual/language/reflection.lagda.rst [hunk 1]
```diff
@@ -822,6 +822,13 @@ Another form of ``unquoteDecl`` is used to declare data types:
 ``m`` is a metaprogram required to declare and define a data type ``x`` and
 its constructors ``c₁`` to ``cₙ`` using ``declareData`` and ``defineData``.
 
+.. note::
+   To debug code generated by ``unquoteDecl`` and ``unquoteDef`` it can be useful to
+   turn on the verbosity flags ``-v tc.unquote.decl:10`` (for type signatures) and
+   ``-v tc.unquote.def:10`` (for definition bodies). This will cause the generated code
+   to be printed on stdout when running from the command line or in the debug buffer
+   when loading from an editor.
```
Comment (andreasabel) [inline doc/user-manual/language/reflection.lagda.rst:830]: This should mention that these verbosities are always available even when Agda is not build with `-f debug`.
I had to look into the Agda sources to clarify this for me.

**Review by andreasabel — CHANGES_REQUESTED** [review]

**Review by andreasabel — APPROVED** [review]
I added a sentence stating that these verbosity flags are available even without building Agda with `-f debug`.

**Comment by UlfNorell** [comment]
I think you meant to write `Unlike other flags`, not `Unless other flags`.

**Comment by andreasabel** [comment]
Oh bugger!

Even in a one-sentence commit I can make mistakes...

**Pull request merged**